src/hardware/serialport/softmodem.cpp

```diff
@@ -355,6 +347,44 @@ char CSerialModem::GetChar(char * & scan) const {
 	return ch;
 }
 
+void CSerialModem::SetModemSpeed(const uint32_t cfg_val) {
+	modem_bps_config   = cfg_val;
+	const auto reported_bps = UpdateConnectString();
+
+	if (modem_bps_config) {
+		LOG_MSG("SERIAL: Port %" PRIu8 " modem will report connection speed "
+		        "of %d bits per second",
+		        GetPortNumber(),
+		        reported_bps);
+	} else {
+		LOG_MSG("SERIAL: Port %" PRIu8 " modem will synchronize reported speed "
+		        "to port baud rate",
+		        GetPortNumber());
+	}
+}
+
+uint32_t CSerialModem::UpdateConnectString() {
+	uint32_t connect_val;
+	if (modem_bps_config != 0) {
+		connect_val = clamp(modem_bps_config,
```

Comment:
Suggest bounding it by `GetPortBaudRate()`, because that's the underlying port's actual baud limit. 

Then the asserts will still sanity checks bounds at the end.

Reply:
Then there's no point in default mode (where modem_bps_config is 0). I can just default modem_bps_config to 57600 and clamp it to port speed.

Reply:
Nice simplification!